fs/xfstests.py

```diff
@@ -221,6 +229,72 @@ def setUp(self):
         self.test_mnt = self.params.get('test_mnt', default='/mnt/test')
         self.disk_mnt = self.params.get('disk_mnt', default='/mnt/loop_device')
         self.fs_to_test = self.params.get('fs', default='ext4')
+        self.run_type = self.params.get('run_type', default='distro')
+
+        if self.run_type == 'upstream':
+            if self.fs_to_test == "ext4":
+                # Build e2fs progs
+                e2fsprogs_dir = os.path.join(self.teststmpdir, 'e2fsprogs')
+                if not os.path.exists(e2fsprogs_dir):
+                    os.makedirs(e2fsprogs_dir)
+                e2fsprogs_url = self.params.get('e2fsprogs_url')
+                git.get_repo(e2fsprogs_url, destination_dir=e2fsprogs_dir)
+                e2fsprogs_build_dir = os.path.join(e2fsprogs_dir, 'build')
+                if not os.path.exists(e2fsprogs_build_dir):
+                    os.makedirs(e2fsprogs_build_dir)
+                os.chdir(e2fsprogs_build_dir)
+                process.run("../configure", verbose=True)
+                build.make(e2fsprogs_build_dir)
+                build.make(e2fsprogs_build_dir, extra_args='install')
+
+            if self.fs_to_test == "xfs":
+                if self.detected_distro.name in ['centos', 'fedora', 'rhel']:
+                    libini_path = process.run("ldconfig -p | grep libini",
+                                              verbose=True, ignore_status=True)
+                    if not libini_path:
+                        # Build libini.h as it is needed for xfsprogs
+                        libini_dir = os.path.join(self.teststmpdir, 'libini')
+                        if not os.path.exists(libini_dir):
+                            os.makedirs(libini_dir)
+                        git.get_repo('https://github.com/benhoyt/inih',
+                                     destination_dir=libini_dir)
+                        os.chdir(libini_dir)
+                        process.run("meson build", verbose=True)
+                        libini_build_dir = os.path.join(libini_dir, 'build')
+                        if os.path.exists(libini_build_dir):
+                            os.chdir(libini_build_dir)
+                            process.run("meson install", verbose=True)
+                        else:
+                            self.fail('Something went wrong while building \
+                                      libini. Please check the logs.')
+                # Build xfs progs
+                xfsprogs_dir = os.path.join(self.teststmpdir, 'xfsprogs')
+                if not os.path.exists(xfsprogs_dir):
+                    os.makedirs(xfsprogs_dir)
+                xfsprogs_url = self.params.get('xfsprogs_url')
+                git.get_repo(xfsprogs_url, destination_dir=xfsprogs_dir)
+                os.chdir(xfsprogs_dir)
+                build.make(xfsprogs_dir)
+                build.make(xfsprogs_dir, extra_args='install')
```

Comment:
Same comment. Would this command install the binaries in /usr/ or /usr/local ?

Reply:
I'll set this explicitly to `/use/local/bin`. _See above comment._